feature_engine/encoding/woe.py

```diff
@@ -30,6 +30,44 @@
 from feature_engine.tags import _return_tags
 
 
+class WoE:
+    def _check_fit_input(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Check that X is dataframe, and y a binary series with values 0 and 1.
+        """
+        X, y = check_X_y(X, y)
+
+        # check that y is binary
+        if y.nunique() != 2:
+            raise ValueError(
+                "This encoder is designed for binary classification. The target "
+                "used has more than 2 unique values."
+            )
+
+        # if target does not have values 0 and 1, we need to remap, to be able to
+        # compute the averages.
+        if y.min() != 0 or y.max() != 1:
+            y = pd.Series(np.where(y == y.min(), 0, 1))
+        return X, y
+
+    def _calculate_woe(self, X: pd.DataFrame, y: pd.Series, variable: str):
+        total_pos = y.sum()
+        inverse_y = y.ne(1).copy()
+        total_neg = inverse_y.sum()
+
+        pos = y.groupby(X[variable]).sum() / total_pos
```

Comment:
How does this work? `y` is a panda series. Doesn't groupby applied to dataframes? Also, should the `by` parameter be a string that is the name of a variable?

Reply:
they work fine. They should not be changed. 

Did you try it out in a jupyter notebook?

Reply:
Perdoname! Soy un boludo!

I did test it in a notebook. The issue was that the dataset had a dependent variable comprised of yes and no, which I mapped to 0 and 1. However, I didn't call the revised/mapped "y" feature when I was testing the WoE class. Ugh!